src/databricks/labs/ucx/workspace_access/groups.py

```diff
@@ -409,19 +409,24 @@ def validate_group_membership(self) -> list[dict]:
         strategy = self._get_strategy(workspace_groups_in_workspace, account_groups_in_account)
         migrated_groups = strategy.generate_migrated_groups()
         mismatch_group = []
-        for groups in migrated_groups:
-            ws_members_set = set([m.get("display") for m in json.loads(groups.members)] if groups.members else [])
-            account_group = account_groups_in_account[groups.name_in_account]
-            ac_members_set = set(
-                [a.as_dict().get("display") for a in account_group.members] if account_group.members else []
-            )
-            set_diff = (ws_members_set - ac_members_set).union(ac_members_set - ws_members_set)
+        retry_on_internal_error = retried(on=[InternalError], timeout=self._verify_timeout)
+        get_account_group = retry_on_internal_error(self._get_account_group)
+        for ws_group in migrated_groups:
+            # Users with the same display name but different email will be deduplicated!
```

Comment:
Isn't that a bug?🐜

Reply:
Comparing by email address would be better. But it's a limitation of the scim groups API. It only returns display name of the user. Email address is not present. We could fetch user info separately but a user can theoretically have multiple email addresses so this might not help.